### PR TITLE
WIP: Add a Tutorial and Wiki section to the documentation

### PR DESCRIPTION
**Pull request by Hombre-x**
This PR has major changes related mostly to the documentation, as it works on adding a tutorial-ish section with a practical demonstration on how to achieve certain things using the library, also with some exercises to do (some trivial some not so much) with their corresponding solution. 

It also adds a small Wiki-like section speaking of the most important methods and how to use them using the extension methods, the static methods and finally, using good ol' Fs2.

This PR adds: 
* Site.scala config containing the `Selection` configurations of the Laika site (the three choices in `mdoc`: `syntax`, `static` and `fs2`)
* The above configuration added in `build.sbt`
* An introductory section talking about the imports, API usage and a *Whats this `IO` thing* section speaking of the advantages  of using it (only referential transparency so far, so more suggestions are welcome!!!!)
* Renaming of the directory in general
* A Wiki section under `docs/files` and `docs/processes`
* A tutorial section under `docs/tutorial`

Whats left to include: 

- [ ] A much more extended and crazy example putting into real practice the library.
- [ ] Of course, any suggestion - change - deletion - addition that you folks recommend!

Lastly, sorry again for the long PR, I can try to split it into smaller ones if that works better for you  :(

### CODE REVIEW

File: build.sbt
Comment (TonioGela) [inline build.sbt:57]: Is this indirection necessary? Given that `SiteConfig.laikaConfig` is comprised of just a few lines, can't we move them here?

File: docs/processes/index.md
Comment (TonioGela) [inline docs/processes/index.md:1]: Let's simply not add this section until we decide to add this feature to the lib.

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md:5]: Can you add for each examples a few words on what you're doing in the example? At the moment this is just a bunch of code examples w/o explanations.

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md]: I would avoid using `FileOs` as a wildcard import, as functions will be added in scope and they might seem "magically" present in scope (IMHO). I'll like to make people repeat the usage of the `FileOs` identifier for each function call, so that they will be able to properly distinguish which are std lib functions and which belong to the lib.

Example.
```diff
-  lines  <- readLines(path)
+  lines  <- FileOs.readLines(path)
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md:3]:
```diff
-this section
+the library
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]: Either

```
In this section you'll see how to create new files and directories, as well as delete existing ones.
```

or

```
(The following feature/Shellfish) lets you to create new files and directories, as well as delete existing ones.
```

File: docs/examples/index.md
Comment (TonioGela) [reply docs/examples/index.md]: This suggestion holds FORALL the code snippet you wrote.

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]:
```suggestion
import cats.syntax.all.*
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md:38]: `// Should return true`

This snippet and all the following ones don't do anything at all. You're not printing the result, not running the IO.

Can you turn the examples into running snippets?

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]:
```suggestion
Creates all the directories in the path, with the default permissions or, with the supplied ones:
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]:
```suggestion
This function creates a temporary file that gets automatically deleted by the operating system. It optionally accepts multiple parameters such as a directory, a prefix (the name of the file), a suffix (the extension of the file) and permissions. It returns the path of the newly created file: 
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]: The functionality itself should be removed from the lib as we don't plan to make newcomers understand what `Resource` does.

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]:
```suggestion
Creates a temporary directory that will eventually be deleted by the operating system. It accepts a few optional parameters like a custom parent directory, a prefix and some permissions. It returns the path to the newly created directory: 
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]: The functionality itself should be removed from the lib.

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md:346]:
```suggestion
Recursively deletes all files or folders, following symbolic links if specified: 
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]:
```suggestion
Very similar to `copy`, but deletes the file in the original destination path. Optionally takes flags as arguments to define its move behaviour:
```

File: docs/files/file_manipulation.md
Comment (TonioGela) [inline docs/files/file_manipulation.md]:
```suggestion
This function checks whether or not a file exists at a specified path:
```

File: docs/index.md
Comment (TonioGela) [inline docs/index.md]: Getting started was good. Also, users won't really "install" the library, they'll use it.

File: docs/index.md
Comment (TonioGela) [inline docs/index.md:13]: Why don't we add an example on how to use the library with scala-cli?

Comment (TonioGela) [inline docs/index.md:13]: You should also replace the following example as it refers to functionality that we'll soon remove.

File: docs/files/index.md
Comment (TonioGela) [inline docs/files/index.md]: Weren't the `WithCharset` methods deprecated?

**Review by TonioGela — CHANGES_REQUESTED** [review]
This review is still a WIP I still need to review a bunch of doc files. I'm "committing" these comments for the moment.

File: docs/files/index.md
Comment (Hombre-x) [reply docs/files/index.md]: Yup, but that was before we talk about the library simplification. Gonna change it right on in the meantime but I need to wait until #77 gots merged so mdoc does not complain about the function signature :crying_cat_face:

File: docs/files/file_manipulation.md
Comment (Hombre-x) [reply docs/files/file_manipulation.md:38]: Not quite, most I can do as I'm aware is compiling them with the `compile-only` flag with `mdoc`. Another option would be to use a library like [mdoc-effect](https://github.com/indoorvivants/mdoc-effect) or create something similar ourselves 🤔

In the meantime, I will start adding the compile only flags to the snippets!

File: build.sbt
Comment (Hombre-x) [reply build.sbt:57]: We totally can, but I segregated the `site` configurations in case we wanted to tune a bit more the site (with custom colors, for example) in the future, so we didn't enlarge the `build.sbt` a lot. Should I segregate them or merge the config anyways and worry about that matter in the future?

File: docs/examples/index.md
Comment (Hombre-x) [reply docs/examples/index.md:5]: Working on it!

File: docs/files/file_manipulation.md
Comment (Hombre-x) [reply docs/files/file_manipulation.md]: I chose the first one

File: docs/files/file_manipulation.md
Comment (Hombre-x) [reply docs/files/file_manipulation.md]: Yup, same as with the `withCharset` variants, going to change that right on as well :D

File: docs/index.md
Comment (Hombre-x) [reply docs/index.md]: You got a point 😂

File: docs/index.md
Comment (Hombre-x) [reply docs/index.md:13]: Honestly, I haven't used scala-cli before. I'll learn it this days and create a small tuto with it :)

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md:18]:
```suggestion
We want to read the file, parse the scores, print them to the console and also append new scores to the file.
```

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md:20]:
```suggestion
In order to perform these operations we first need to create a representation of the score in Scala:
```

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md:26]:
```suggestion
```

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md:28]:
```suggestion
and then a function that parses a string into a `Either[Throwable, Score]`:
```

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md:38]:
```suggestion
The method is going to return a `Left` or a `Right` according to to the outcome of the score parsing. The function consists in two steps:
```

File: docs/examples/index.md
Comment (TonioGela) [inline docs/examples/index.md:43]:
```suggestion
We can now create our script: 
```

**Comment by Hombre-x** [comment]
So, as suggested by my fellow contributors, this PR shall be separated into smaller ones for the sanity of us all. 

I will not close this PR just yet, just to have it as a track of the rest of the pull requests that need approval, just to be in line with the status of this one. 

This is the list of the remaining PRs that need merging:

- [x] #81 (needs merging first because of dependencies)
- [ ] #82
- [ ] #85 Tutorial part 1: Using files API here and there
- [ ] #86 Tutorial part 2: Contact Manager (possibly also scala-cli tutorial)
- [ ] #84 Wiki: Overview of the methods and quick explanations of the API
- [ ] #83 Introduction: Imports, IO and general things (possibly future FAQ section)

Till then, I shall meet thee upon the path to fulfill this.

**Pull request closed without merging**
